x/qgb/abci_test.go

```diff
@@ -111,13 +111,88 @@ func TestSetDataCommitment(t *testing.T) {
 	input, ctx := testutil.SetupFiveValChain(t)
 	qk := input.QgbKeeper
 
-	input.Context = ctx.WithBlockHeight(int64(qk.GetDataCommitmentWindowParam(ctx)))
-	vs, err := qk.GetCurrentDataCommitment(ctx)
+	ctx = ctx.WithBlockHeight(int64(qk.GetDataCommitmentWindowParam(ctx)))
+	dc, err := qk.GetCurrentDataCommitment(ctx)
+	require.NoError(t, err)
+	err = qk.SetAttestationRequest(ctx, &dc)
+	require.NoError(t, err)
+
+	require.Equal(t, uint64(1), qk.GetLatestAttestationNonce(input.Context))
+}
+
+// TestGetDataCommitment This test will test the create of data commitment ranges
+// in the event of the data commitment window changing via an upgrade or a gov proposal.
+// The test goes as follows:
+//   - Start with a data commitment window of 400
+//   - Get the first data commitment, its range should be: [1, 400]
+//   - Get the second data commitment, its range should be: [401, 800]
+//   - Shrink the data commitment window to 101
+//   - Get the third data commitment, its range should be: [801, 901]
+//   - Expand the data commitment window to 500
+//   - Get the fourth data commitment, its range should be: [902, 1401]
+func TestGetDataCommitment(t *testing.T) {
+	input, ctx := testutil.SetupFiveValChain(t)
+	qk := input.QgbKeeper
+
+	dcWindow := qk.GetDataCommitmentWindowParam(ctx)
+
+	// get the first data commitment
+	ctx = ctx.WithBlockHeight(int64(dcWindow))
+	dc1, err := qk.GetCurrentDataCommitment(ctx)
+	require.NoError(t, err)
+	require.Equal(t, uint64(1), dc1.BeginBlock)
+	require.Equal(t, uint64(dcWindow), dc1.EndBlock)
+	require.Equal(t, uint64(1), dc1.Nonce)
+	// set the first data commitment
+	err = qk.SetAttestationRequest(ctx, &dc1)
+	require.Nil(t, err)
+
+	// get the second data commitment
+	ctx = ctx.WithBlockHeight(int64(dcWindow) * 2)
+	dc2, err := qk.GetCurrentDataCommitment(ctx)
 	require.Nil(t, err)
-	err = qk.SetAttestationRequest(ctx, &vs)
+	require.Equal(t, dc1.EndBlock+1, dc2.BeginBlock)
+	require.Equal(t, dc1.EndBlock+dcWindow, dc2.EndBlock)
+	require.Equal(t, uint64(2), dc2.Nonce)
+	// set the second data commitment
+	err = qk.SetAttestationRequest(ctx, &dc2)
 	require.Nil(t, err)
 
-	require.Equal(t, uint64(1), qk.GetLatestAttestationNonce(ctx))
+	// shrink the data commitment window
+	genesis := types.DefaultGenesis()
+	newShrinkedDCWindow := dcWindow - 299 // 101, since the default one is 400
```

Comment:
[question] is there a rationale to use `101` instead of `100`? I would expect `100` to produce friendlier (to the human eye) ranges to verify

```suggestion
	newShrinkedDCWindow := 101
```

Reply:
100 can give ranges that are similar to the ones for 400, if the upgrade height is 1200 for example. That's why I used 101 so that we can see the difference.
Also, 101 is the lower bound for data commitment windows, but it's unrelated here.

Reply:
> That's why I used 101

Makes sense, thanks!

> Also, 101 is the lower bound for data commitment windows, but it's unrelated here.

Where is this lower bound specified? Why is `101` the lower bound?

Reply:
The lower bound is specified in here:
https://github.com/celestiaorg/celestia-app/blob/b2e8a72c76d40003acb93b988debe92cd110ae99/x/qgb/types/genesis.go#L54-L55

Now that I am checking, it's 100 :D 

The reason for that is if we start sending attestations for lower ranges, we will need to relay more transactions, which means spending more money. Also, since we're able to batch lots of blocks in the same data commitment, the more we batch, the easier it is to find the commitments and prove them.

Reply:
Thanks for sharing!

1. should the `100` be extracted to a constant?
2. I don't see why specifying a data commitment window of `99` would result in the error: `"invalid average EVM block time, too short for latency limitations"` because the data commitment window is not related to the average EVM block time. I would expect an error message like

```go
const MinimumDataCommitmentWindow = 100

func validateDataCommitmentWindow(i interface{}) error {
	val, ok := i.(uint64)
	if !ok {
		return fmt.Errorf("invalid parameter type: %T", i)
	} else if val < uint64(MinimumDataCommitmentWindow) {
		return fmt.Errorf("data commitment window %v must be >= minimum data commitment window %v", val, MinimumDataCommitmentWindow)
	}
	return nil
}
```

Reply:
> should the 100 be extracted to a constant?

Could be. But I don't see a reason for now why

> I don't see why specifying a data commitment window of 99 would result in the error: "invalid average EVM block time, too short for latency limitations" because the data commitment window is not related to the average EVM block time. I would expect an error message like

I'll get to it  and update the message https://github.com/celestiaorg/celestia-app/issues/1706, thanks

Reply:
Yea sorry this comment is not blocking this PR. Good idea to create a separate issue. Resolving!